File: internal-frontend/src/app/dto/ConsumerGroupDto.ts
import {WiringMode} from "./WiringMode";
import {ConsumerMode} from "./ConsumerMode";
import {BatterySystemDto} from "./BatterySystemDto";
import {Dto} from "./Dto";

export class ConsumerGroupDto extends Dto {
  name: string = '';
  gatewayId: string = '';
  voltage = 0;
  systems: BatterySystemDto[] = [];
  mode: ConsumerMode = ConsumerMode.None;
  wiring: WiringMode = WiringMode.Unknown;
}

File: StorageApi/src/main/kotlin/org/vpp/storageApi/consumerGroup/ConsumerGroupDto.kt
package org.vpp.storageApi.consumerGroup

import kotlinx.serialization.Serializable
import org.vpp.storage.consumerGroup.ConsumerMode
import org.vpp.storage.consumerGroup.WiringMode
import org.vpp.storageApi.batterySytem.BatterySystemDto
import org.vpp.storageApi.dto.Dto

@Serializable
class ConsumerGroupDto : Dto() {
    lateinit var name: String
    var voltage: Double = 0.0
    lateinit var mode: ConsumerMode
    lateinit var wiring: WiringMode
    var systems: List<BatterySystemDto> = emptyList()
}
